### ISSUE TITLE
Intel AppImage issues

### ISSUE THREAD

**Issue by Miepee** [Lojemiru/AM2R-Autopatcher-Linux#19]
For some reason, intel iGPUs segfault on the AppImage.
This makes it (nearly) impossible for such users to play the native AM2R Linux version:

Not exactly sure *why* this happens, as the intel drivers *are* included there. 
I don't have an intel machine to try this out on, sadly.

**Comment by ludvigng** [comment]
There's probably some problem with libgl, mounting the appimage with ./AM2R.AppImage --appimage-mount and then entering the mounted image in /tmp and manually running the app with LIBGL_DRI3_DISABLE=1 ./AppRun makes it start.

**Comment by Miepee** [comment]
So basically `./AM2R.AppImage --appimage-mount-and-run` ?
EDIT: Nevermind, just saw about that environment variable.
Can you confirm, if the native binary has the segfault problem as well, and is fixable with that environment variable as well?
Also, does the AppImage run, if you just run it like this `LIBGL_DRI3_DISABLE=1 ./AM2R.AppImage`?

**Comment by ludvigng** [comment]
> Can you confirm, if the native binary has the segfault problem as well, and is fixable with that environment variable as well?

It didn't, but given the workarounds I had to do to get it running (libcurl3 and openssl 1.0) it's really unfeasable on ubuntu.
> Also, does the AppImage run, if you just run it like this `LIBGL_DRI3_DISABLE=1 ./AM2R.AppImage`?

Yeah it does. Don't know why I didn't think of that at first.

**Comment by Miepee** [comment]
Just out of curiosity, can you give me a neofetch for spec information?

**Comment by ludvigng** [comment]
![Skärmbild från 2021-02-19 18-24-54](https://user-images.githubusercontent.com/50339757/108539099-d282e800-72df-11eb-97a9-d7a875f19535.png)

It's a dual GPU laptop, currently running on the intel card. Running the game with discrete graphics still segfaults though.

**Comment by Miepee** [comment]
Thank you!
I'll try to get some people to test that fix!

**Comment by Miepee** [comment]
Also @ludvigng you don't happen to know if that's maybe an Ubuntu/Debian derivatives specific problem?
So far, I only encountered these problems on these machines.
Me and my "test subjects" mostly use Arch, and we never got any problems there.

**Comment by ludvigng** [comment]
I honestly have no idea, sorry. I can see if I can setup a test environment later, but no promises. :)

**Comment by Miepee** [comment]
It's understandable if it's not possible, but I still wish you good luck with that!
In the meantime, I'll try this test on my "test subjects"

**Comment by ludvigng** [comment]
Did the simplest test possible, booted a live environment manjaro and just copied the AppImage over, and it worked fine out of the box. So yeah there's something going on with ubuntu it seems.

**Comment by Miepee** [comment]
Heya, @ludvigng can you try the intel-ubuntu-fix branch, and see if the issue is resolved there?

**Comment by ludvigng** [comment]
Tried it in a live image, following the new instructions (enable universe repo, installing python3, xdelta3, enabling multilib and installing libc6:i386 libncurses5:i386 libstdc++6:i386 and libopenal1:i386). It now runs!

There's still a problem with audio, but that's probably due to me trying it on an live environment and unrelated to this issue (I fixed it on my old system, but I don't remember how and have since switched distro, I suspect it's just a mismatch due to new lib32 stuff interacting with the old 20.10 live system). :)

**Comment by Miepee** [comment]
Thanks for testing! I'll PR it then!
I assume the sound problem might've been me, I made a commit in master which would hopefully fix that. It's either a conflicting sound library in the appimage, or a missing libopenal1 install.

**Comment by ludvigng** [comment]
No prob!

I made sure libopenal1:i386 was installed in the live environment, but if I remember correctly how I made it work was to install libpulse0:i386 on my old system. I did also try that in the live env, but it was already installed, so I think it's just the live env being the way it is. Either way people probably won't patch this from a live env, so it should probably run fine on an actual system now.
